### PR TITLE
Fixup the docs about docs

### PR DESCRIPTION
**Pull request by evantahler**
* Update docs for developing docs.airbyte.com - [it's a 1-pager now](https://github.com/airbytehq/airbyte/blob/433c6304f53d18e9a642a38105e004d8f7117cec/docusaurus/README.md)
* Remove old docs that aren't right any more

### CODE REVIEW

**Comment by evantahler** [comment]
Ping @supertopher or @erohmensing!

File: docusaurus/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+## Developing Locally
+
+```bash
+yarn start # any changes will automatically be reflected in your browser!
+```
+
+## Making Changes
+
+All the content for docs.airbute.com lives in the `/docs` directory in this repo. All files are markdown. Make changes or add new files, and you should see them in your browser!
```
Comment (erohmensing) [inline docusaurus/README.md:31]: New files will not actually be shown - users must add them to the sidebars.js (and maybe rebuild?  not sure) to get them to show up

File: docusaurus/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
+# Documentation and Docusaurus
+
+We use [docusaurus](https://docusaurus.io) for consistent process, in `Airbyte` **no website is generated**.
+Functionally this is a very fancy **linter**
+
+Running the build process will **check for broken links**, please read the output and address
+any broken links that you are able to do.
```
Comment (erohmensing) [inline docusaurus/README.md:7]: Just wondering, is this something we rely on people who build the docs to check and fix, or do we have CI for it? I know we've had issues with broken links in the past

**Review by erohmensing — APPROVED** [review]
TY! So many places for information 😄 very helpful to consolidate

File: docusaurus/README.md (hunk 2 above)
Comment (evantahler) [reply docusaurus/README.md:7]: CI will tell you!

Comment (erohmensing) [reply docusaurus/README.md:7]: Sweet!

Comment (natikgadzhi) [reply docusaurus/README.md:7]: There's a caveat: Docusaurus link checker only checks for internal links to markdown files. If there's a link to a website with a full URL, even our docs site, I don't know if it's going to be checked as a build step. I'll take a look.

**Pull request merged**